### PR TITLE
ADLR v2.0.0 slim down public api

### PR DESCRIPTION
**Pull request by ianhecker**
With some restructuring of directories and an `api/` pkg for interface and type wrappers, most of the API has been pushed into `internal/`

### CODE REVIEW

File: internal/auditor.go [hunk 1]
```diff
@@ -8,18 +8,12 @@ const NonWhitelistedLicenseErr = "non-whitelisted license: "
...
-func MakeLicenseAuditor() LicenseAuditor {
-	whitelist := MakeLicenseWhitelist()
-	return MakeLicenseAuditorFromRaw(whitelist)
-}
-
-// Create a LicenseAuditor from specified values
-func MakeLicenseAuditorFromRaw(
-	whitelist LicenseWhitelist,
+// Create a LicenseAuditor from specified Whitelist
+func MakeLicenseAuditor(
```
Comment (ianhecker) [inline internal/auditor.go:15]: this is prep work for specifying a whitelist via cmd flag

File: api/whitelist.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import "github.com/blocky/adlr/internal"
+
+// This list is of SPDX License Identifiers, the standard
+// used by the text-mining package:
+// github.com/go-enry/go-license-detector/v4
+// and contains licenses deemed automatically fulfillable.
+// To add to this list, see:
+// https://spdx.org/licenses/
+// for license identifiers
+var DefaultWhitelist = []string{
```
Comment (ianhecker) [inline api/whitelist.go:12]: prep work for whitelist specification via cmd flag

File: internal/licenselock.go [hunk 3]
```diff
@@ -49,33 +49,33 @@ func MakeLicenseLockFromRaw(
 // Lock DependencyLocks derived by adlr of a project's golang
 // module dependencies into a lock file for vcs
 func (lock LicenseLock) Lock(
-	locks []DependencyLock,
+	locks ...DependencyLock,
```
Comment (ianhecker) [inline internal/licenselock.go:52]: moving to variadic to match the rest of the functions

**Review by ianhecker — COMMENTED** [review]
@dlm It's mostly renames, but let me know if you want to joint review. And what you think of the api pkg idea

**Review by dlm — APPROVED** [review]
This looks good!!

**Pull request merged**